thenewboston_node/business_logic/storages/file_system.py

```diff
@@ -92,10 +98,18 @@ class FileSystemStorage:
     Compressing / decompressing storage for capacity optimization
     """
 
-    def __init__(self, base_path: Union[str, Path], compressors=tuple(COMPRESSION_FUNCTIONS), temp_dir='.tmp'):
+    def __init__(
+        self,
+        base_path: Union[str, Path],
+        compressors=tuple(COMPRESSION_FUNCTIONS),
+        temp_dir='.tmp',
+        always_compress=False
```

Comment:
`always_compress=True` - would be backward compatible. Why did you change it?

Why we need this argument at all?

Reply:
> always_compress=True - would be backward compatible

No, `always_compress=False` is backward compatible. See this 
https://github.com/thenewboston-developers/thenewboston-node/pull/373/files#diff-115245d6e6f9c1688b3e587bea990a12c21d557cf572a066700cd3e806ce59a0L194
By default, files are compressed only if the compressed file size is less than non-compressed. It happens usually if the file size is small, i.e. contains only one block or small blockchain state. In tests, for example. As we call `generate_blockchain` in tests, blockchain is dynamic. And sometimes files are gzipped, sometimes not. Which makes some tests flapping. So I added this argument to fix it and have more control over compression. Add to check that `blockchain_state.meta` has the right file path which is checked by this test https://github.com/thenewboston-developers/thenewboston-node/pull/373/files#diff-92b3fe228a2ee98ab3ddeb5bfb117dbf65c7a2bd57aa8b0f787fa5fcb720a47dR9

Should we always compress the files then?

Reply:
I believe we should add features to implementation that are need only for testing (if this is possible). Can we instead patch storage with empty list of compressors? This should make test stable.

But if you want to test compressed files to have better test coverage the better solution would to consider no compression as just another type compression that does nothing. And after this patch the list of compressors with only one compressor (any of your choice).